### ISSUE TITLE
Gradual Unfreezing

### ISSUE THREAD

**Issue by Robert60CB** [UKPLab/sentence-transformers#260]
Hi, I'm back. Once again I feel this library truly is one of the best ways to learn about NLP - after plugging and playing some of the models - and then dissecting the code underlying those models including pooling, I've really learned so much. 

I understand to fine tune on a task like NLI or STS, a head is placed on top of the model. I've experimented with LR, weight decay, schedule etc but am wondering if there is an argument to do gradual unfreezing? 

I encountered that concept through Jeremy Howard's ULMFIT paper. If there isn't, I'll try venture and see if I can implement it. 

But, as everyone here is more experienced than I and has a lot more context - is it worth exploring gradual unfreezing when fine tuning? Or am I better off learning and experimenting with other concepts like catastrophic forgetting, WK Pooling etc?

**Comment by nreimers** [comment]
Hi @Robert60CB 
happy to hear that :)

This is not yet implemented. But I think it can be quite interesting.

The challenge in fine-tuning BERT is to transform the network such that it produces sensible vectors that can be compared using cosine similarity. Out of the box, BERT does not great appropriate embeddings.

However, as we have only little data for fine-tuning with NLI/STS, we do not want to change the BERT model too much, i.e., the language information in BERT should not be lost by our fine-tuning. Here, gradual unfreezing might be helpful.
